tracing/tracer.go

```diff
@@ -100,3 +103,17 @@ func (jr *jaegerReporter) UpdateGauge(name string, tags map[string]string, value
 func (jr *jaegerReporter) RecordTimer(name string, tags map[string]string, d time.Duration) {
 	jr.reporter.AllocateTimer(name, tags).ReportTimer(d)
 }
+
+// WithSpan is used for generating a span to be used in testing
+func WithSpan(t *testing.T, log ulog.Log, f func(opentracing.Span)) {
```

Comment:
I don't know. I will be very confused if godoc said `WithSpan` takes in a `testing.T`, but the comment explains though. Still something I would try to avoid show up in API.